File: solutions/javascript/resistor-color/1/resistor-color.js

export const colorCode = (color) => {
    // Using linear search
    for (var i = 0; i < COLORS.length; i++) {
        if (color === COLORS[i]) {
            return i;
        }
    }
};

export const COLORS = ["black","brown","red","orange","yellow","green","blue","violet","grey","white"];

File: solutions/go/resistor-color/1/resistor_color.go
package resistorcolor

// Colors returns the list of all colors.
func Colors() []string {
	colors := []string{"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"}

	return colors
}

// ColorCode returns the resistance value of the given color.
func ColorCode(color string) int {
	var i int

	colors := Colors()
	for k, v := range colors {
		if v == color {
			i = k
			break
		}
	}

	return i
}
